modules/benchmarks/src/lib.rs

```diff
@@ -1,183 +1,359 @@
 #![allow(clippy::too_many_arguments)]
 use spacetimedb::{println, spacetimedb};
+use std::hint::black_box;
+
+// the following piece of code must remain synced with `crates/bench/src/schemas.rs`
+// unfortunately, you can't just copy it.
+
+// ---------- SYNCED CODE ----------
+
+#[spacetimedb(table)]
+pub struct UniquePerson {
+    #[unique]
+    id: u32,
+    name: String,
+    age: u64,
+}
+
+#[spacetimedb(table)]
+pub struct NonUniquePerson {
+    id: u32,
+    name: String,
+    age: u64,
+}
 
 #[spacetimedb(table)]
-pub struct Person {
+#[spacetimedb(index(btree, name = "id", id))]
+#[spacetimedb(index(btree, name = "name", name))]
+#[spacetimedb(index(btree, name = "age", age))]
+pub struct MultiIndexPerson {
+    id: u32,
     name: String,
+    age: u64,
+}
+
+#[spacetimedb(table)]
+pub struct UniqueLocation {
+    #[unique]
+    id: u32,
+    x: u64,
+    y: u64,
+}
+
+#[spacetimedb(table)]
+pub struct NonUniqueLocation {
+    id: u32,
+    x: u64,
+    y: u64,
+}
+
+#[spacetimedb(table)]
+#[spacetimedb(index(btree, name = "id", id))]
+#[spacetimedb(index(btree, name = "x", x))]
+#[spacetimedb(index(btree, name = "y", y))]
+pub struct MultiIndexLocation {
+    id: u32,
+    x: u64,
+    y: u64,
+}
+// ---------- / SYNCED CODE ----------
+
+#[spacetimedb(reducer)]
+pub fn empty() {}
+
+// ---------- insert ----------
```

Comment:
One kind of documentation would be super useful for this, and that's the framework that you used to derive all of these. Suppose we add some new operation, I'd love to then be able to methodically add the "correct" set of new things here.